src/Parser.hs

```diff
@@ -245,26 +243,26 @@ bops = [
   ("<", Lt)
   ]
 
-parseBOp :: Parser (BOp)
-parseBOp = let (p:ps) = map (uncurry translate) bops in foldl (<|>) p ps
-
   -- TODO no bitwise ops in here yet
 
+parseBlock :: Parser [Expr]
+parseBlock = braces $ many1 parseExpr
+
 -- | Parse an expr
 parseExpr' :: Parser (Expr)
 parseExpr' =
   DT.trace "parsing expression..." $
-  Mut <$> (reserved "mut" *> parseType) <*> lowIdentifier <*> (reserved "=" *> parseExpr) <*> parseExpr
+  Mut <$> (reserved "mut" *> parseType) <*> lowIdentifier <*> (reserved "=" *> parseExpr)
   <|>
-  Const <$> (reserved "const" *> parseType) <*> lowIdentifier <*> (reserved "=" *> parseExpr) <*> parseExpr
+  Const <$> (reserved "const" *> parseType) <*> lowIdentifier <*> (reserved "=" *> parseExpr)
   <|>
-  Update <$> (reserved "set" *> lowIdentifier) <*> parseExpr <*> parseExpr
+  Update <$> (reserved "set" *> lowIdentifier) <*> parseExpr
   <|>
-  Out <$> (reserved "out" *> lowIdentifier) <*> parseExpr <*> parseExpr
+  Out <$> (reserved "out" *> lowIdentifier) <*> parseExpr
   <|>
-  Branch <$> (reserved "if" *> parseExpr) <*> (reserved "then" *> (reserved "{" *> parseExpr <* reserved "}")) <*> (reserved "else" *> (reserved "{" *> parseExpr <* reserved "}"))
+  Branch <$> (reserved "if" *> parseExpr) <*> (reserved "then" *> parseBlock) <*> (reserved "else" *> parseBlock)
   <|>
-  For <$> (reserved "for" *> parseExpr) <*> (reserved "do" *> return Nothing) <*> (reserved "{" *> parseExpr <* reserved "}") <*> parseExpr
+  For <$> (reserved "for" *> parseExpr) <*> (consume ("do", Nothing)) <*> parseBlock
```

Comment:
This is a bit of hack still. It's just a for loop w/ no referencing variable in the loop itself, but the restructuring is fine here. We just need to take care of it before we finish.